packages/lib/src/components/Giftcard/components/GiftcardComponent.tsx

```diff
@@ -107,55 +111,17 @@ class Giftcard extends Component<GiftcardComponentProps> {
-                                focused={focusedElement === 'encryptedCardNumber'}
-                                onFocusField={() => setFocusOn('encryptedCardNumber')}
-                                dir={'ltr'}
-                                name={'encryptedCardNumber'}
-                            >
-                                <DataSfSpan
-                                    encryptedFieldType="encryptedCardNumber"
-                                    data-info='{"length":"15-32", "maskInterval":4}'
-                                    className={classNames({
-                                        'adyen-checkout__input': true,
-                                        'adyen-checkout__input--large': true,
-                                        'adyen-checkout__card__cardNumber__input': true,
-                                        'adyen-checkout__input--error': getCardErrorMessage(sfpState),
-                                        'adyen-checkout__input--focus': focusedElement === 'encryptedCardNumber'
-                                    })}
-                                />
-                            </Field>
-
-                            {props.pinRequired && (
-                                <Field
-                                    label={i18n.get('creditCard.pin.title')}
-                                    classNameModifiers={['pin', '30']}
-                                    errorMessage={sfpState.errors.encryptedSecurityCode && i18n.get(sfpState.errors.encryptedSecurityCode)}
-                                    focused={focusedElement === 'encryptedSecurityCode'}
-                                    onFocusField={() => setFocusOn('encryptedSecurityCode')}
-                                    dir={'ltr'}
-                                    name={'encryptedSecurityCode'}
-                                >
-                                    <DataSfSpan
-                                        encryptedFieldType="encryptedSecurityCode"
-                                        data-info='{"length":"3-10", "maskInterval": 0}'
-                                        className={classNames({
-                                            'adyen-checkout__input': true,
-                                            'adyen-checkout__input--large': true,
-                                            'adyen-checkout__card__cvc__input': true,
-                                            'adyen-checkout__input--error': sfpState.errors.encryptedSecurityCode,
-                                            'adyen-checkout__input--focus': focusedElement === 'encryptedSecurityCode'
-                                        })}
-                                    />
-                                </Field>
-                            )}
-                        </div>
-                    )}
+                    render={({ setRootNode, setFocusOn }, sfpState) =>
+                        h(this.props.fieldsLayoutComponent, {
```

Comment:
Hey @m1aw, maybe a fragment (instead of calling `h`) would make this one a bit easier to understand, what do you think?

Reply:
Fixed